src/component/tag-types/form-tag-type-component.js

```diff
@@ -0,0 +1,116 @@
+            dirty: false,
+            currentLegalValue: '',
+        };
+    }
+
+    static getDerivedStateFromProps(props, state) {
+        if (!state.tagType && props.tagType.name) {
+            return { tagType: props.tagType };
+        } else {
+            return null;
+        }
+    }
+
+    validateName = async name => {
+        const { errors } = this.state;
+        const { validateName } = this.props;
+        try {
+            await validateName(name);
+            errors.name = undefined;
+        } catch (err) {
+            errors.name = err.message;
+        }
+        this.setState({ errors });
+    };
+
+    setValue = (field, value) => {
+        const { tagType } = this.state;
+        tagType[field] = value;
+        this.setState({ tagType, dirty: true });
+    };
+
+    onCancel = evt => {
+        evt.preventDefault();
+        this.props.history.push('/tag-types');
+    };
+
+    onSubmit = async evt => {
+        evt.preventDefault();
+        const { tagType } = this.state;
+        await this.props.submit(tagType);
+        this.props.history.push('/tag-types');
+    };
+
+    render() {
+        const { tagType, errors } = this.state;
+        const { editMode } = this.props;
+        const submitText = editMode ? 'Update' : 'Create';
+        return (
+            <Card shadow={0} className={commonStyles.fullWidth} style={{ overflow: 'visible' }}>
+                <CardTitle style={{ paddingTop: '24px', paddingBottom: '0', wordBreak: 'break-all' }}>
+                    {submitText} Tag type
+                </CardTitle>
+                <CardText>Tag types allows you to group tags together in the management UI</CardText>
+                <form onSubmit={this.onSubmit}>
+                    <section style={{ padding: '16px' }}>
+                        <Textfield
+                            floatingLabel
+                            label="Name"
+                            name="name"
+                            placeholder="url-friendly-unique-name"
+                            value={tagType.name}
+                            error={errors.name}
+                            disabled={editMode}
+                            onBlur={v => this.validateName(v.target.value)}
+                            onChange={v => this.setValue('name', trim(v.target.value))}
+                        />
+                        <Textfield
+                            floatingLabel
+                            style={{ width: '100%' }}
+                            label="Description"
+                            name="description"
+                            placeholder="Some short helpful descriptive text"
+                            rows={1}
+                            error={errors.description}
+                            value={tagType.description}
+                            onChange={v => this.setValue('description', v.target.value)}
+                        />
+                        <Textfield
```

Comment:
I am leaning towards not to expose the icon functionality for custom tags in the UI yet, as I feel there is a lot of things to figure our how this should work.

(We have used a defined list of all icon names from material icons under application, but I do not feel this is a "perfect" solution either).

Reply:
what do you think?